ls1intum/Artemis
# [Bugfix] Metis postings render markdown correctly

### Checklist

### Motivation and Context


- Solves #3576
- Create a dedicated `postings-markdown-editor` to avoid code duplication
- Adapts the default commands (shown formatting options) for the postings editor, which are not limited to the most important ones

### Description

The array of allowed html tags and attributes in `answer-post.component.ts` and `post.component.ts` was too restrictive. Almost every formatting feature the markdown editor was offering got lost. 

By extending `allowedHtmlTags` and `allowedHtmlAttributes`, the browser is able to render, what is promised by the markdown preview without introducing possible XSS vulnerabilities. 

Headings ("#h1", "##h2") rely on the attribute `id` to be allowed (which we do not want), and can therefore not be rendered correctly.

The "format-short-cut"-bar of the markdown editor was reduced to formats that are rendered correctly and most probably useful. (Un-)Ordered lists are also rendered correctly but are, for the sake of simplicity, excluded from the "format-short-cut"-bar. Simply using "-" (for unordered lists) and "1." (for ordered lists) should be intuitive.

<img width="373" alt="Bildschirmfoto 2021-07-07 um 16 59 21" src="https://user-images.githubusercontent.com/41366522/124782656-aee5d480-df44-11eb-95f0-f186bb5d57dd.png">

### Steps for Testing


1. Log in to Artemis
2. Navigate to a course with postings enabled and within that course to an exercise or lecture page -> see discussion section on the right
3. Create some posts and answer posts using the markdown editor features, e.g. such as shown in the input screenshot etc.
4. Check if the (answer) post is displayed as shown in the preview after saving it.

### Screenshots


**Input**:
<img width="407" alt="Bildschirmfoto 2021-07-07 um 17 02 35" src="https://user-images.githubusercontent.com/41366522/124783153-2451a500-df45-11eb-96ec-87f286be172e.png">

**Before**:
<img width="578" alt="Bildschirmfoto 2021-07-07 um 17 03 11" src="https://user-images.githubusercontent.com/41366522/124783243-37fd0b80-df45-11eb-9236-6f722d66335e.png">

**After**:
<img width="411" alt="Bildschirmfoto 2021-07-07 um 17 03 40" src="https://user-images.githubusercontent.com/41366522/124783346-48ad8180-df45-11eb-91a3-802c13babd21.png">
